### ISSUE TITLE
Bug: transparent images overlap on each other

### ISSUE THREAD

**Issue by ccrowles** [riyasy/FlyPhotos#16]
Basically when scrolling through photos, some transparent images start to "stack" on each other.

To better explain what i mean, I made a gif showing what happens when I scroll through my folder's images:

https://github.com/user-attachments/assets/1f9f98fd-1b6b-4a97-ad02-100e3036a48e

**Comment by riyasy** [comment]
Does this happen always happen with partially transparent image files?

**Comment by riyasy** [comment]
I checked with some random downloaded partially transparent PNGs from internet and could not reproduce the issue. If you can share some sample images with which I can test, it would be helpful.

**Comment by ccrowles** [comment]
Certainly! I can put the transparent pngs that were giving me the error in the video in a zipped archive:
[pngs with issues.zip](https://github.com/user-attachments/files/16710560/pngs.with.issues.zip)

**Comment by riyasy** [comment]
I could not reproduce the issue in my PC. Can you check using the latest version released today ?

**Comment by ccrowles** [comment]
Uh... I installed the updated version but now FlyPhotos doesn't start at all. 


https://github.com/user-attachments/assets/e86c8b1e-da62-463e-adf3-5ef93f3d805e

**Comment by riyasy** [comment]
Can you uninstall and do a clean install ?
Did you update or was it a clean install?

**Comment by ccrowles** [comment]
I now updated and did a clean install, FlyPhotos works but the issue is not fixed. Those images still overlap...

![Screenshot_2024-08-24_16-01-05_145](https://github.com/user-attachments/assets/8fb15aef-5f0e-49b5-becf-b34e2b115b9f)

**Comment by riyasy** [comment]
Thanks for checking again.  I shall check again in the same order as yours..

**Comment by riyasy** [comment]
Are you using some desktop customization tools? If yes can you check with them turned off if possible?

**Comment by ccrowles** [comment]
I'm not sure what exactly would you define as deskop customization tools, the only thing that I can think of that interacts with windows explorer is [Nilesoft Shell](https://nilesoft.org), which changes the context menu. But it happens even when disabling it.

**Comment by riyasy** [comment]
I saw a transparent taskbar, decorative calendar and weather widgets etc in the screenshot and thought may be some tool was used to customize the desktop.

**Comment by ccrowles** [comment]
Oh you mean Rainmeter? Those are Rainmeter widgets. Even closing rainmeter, the issue remains.

**Comment by riyasy** [comment]
Ok. I shall check on my side again. Thanks.

**Comment by ccrowles** [comment]
Ok! I also disabled TaskbarX, which was what affected the taskbar transparency (completely forgot about that), but the issue still persists even when disabling it.

**Comment by prm88** [comment]
Lol, I'm having the same problem (accidentally came to this thread before I even installed the program)
Windows is only like 1-2 days freshly installed. All drivers/dependencies are installed. Intel/Nvidia gtx 1070 level computer.

Opaque pictures displays - perfectly.
But in the folder “C:\Program Files\RYFTools\Fly” icons overlap each other.
Checked with other transparent images - same problem.

Switching the “window background” setting didn't help.
There are no errors in the log file.
I decided to switch “Windows > Personalize > Colors > Transparency effects” for fun - didn't help.

https://postimg.cc/zycD8tWB
https://postimg.cc/1gVmtGcK
https://postimg.cc/zbmJRyxY

**Comment by riyasy** [comment]
> Lol, I'm having the same problem (accidentally came to this thread before I even installed the program) Windows is only like 1-2 days freshly installed. All drivers/dependencies are installed. Intel/Nvidia gtx 1070 level computer.
> 
> Opaque pictures displays - perfectly. But in the folder “C:\Program Files\RYFTools\Fly” icons overlap each other. Checked with other transparent images - same problem.
> 
> Switching the “window background” setting didn't help. There are no errors in the log file. I decided to switch “Windows > Personalize > Colors > Transparency effects” for fun - didn't help.
> 
> https://postimg.cc/zycD8tWB https://postimg.cc/1gVmtGcK https://postimg.cc/zbmJRyxY

I understand. But I am not able to reproduce the issue in both Windows 11 and Windows 10 machines at my disposal.
Since directx is being used for rendering, I suspect if this has got anything to do with Graphics card and all.
Anyway I shall check the drawing code once again and do some cleanup between image switches.
https://postimg.cc/vc801Xvd
https://postimg.cc/5XTgRJWx

**Comment by SABERBIRD** [comment]
This does not depend on specific images. I have several different ones, and all of them have the same issue.

Sometimes this bug does not appear from the first viewing of the image, but only after scrolling forward and then back.

Also, in my video, you can see that not only transparent images stack, but opaque ones as well. Not all of them, but only the last one rendered.

I use decorative programs, but even after disabling them, the bug remained (I did not try uninstalling the programs).

- Lively Wallpaper - animated desktop wallpaper
- TranslucentTB - transparent taskbar

https://github.com/user-attachments/assets/c686d6cb-699c-443a-a0f3-1571a5a6d06a

_Win 11 (without TPM 2.0)_
_nvidia geforce GTX 1070_
_amd ryzen 5 1500X quad-core_
_Fly Photos 2.3.2_

**Comment by riyasy** [comment]
> This does not depend on specific images. I have several different ones, and all of them have the same issue.
> 
> Sometimes this bug does not appear from the first viewing of the image, but only after scrolling forward and then back.
> 
> Also, in my video, you can see that not only transparent images stack, but opaque ones as well. Not all of them, but only the last one rendered.
> 
> I use decorative programs, but even after disabling them, the bug remained (I did not try uninstalling the programs).
> 
> * Lively Wallpaper - animated desktop wallpaper
> * TranslucentTB - transparent taskbar
> 
>  Fly.Photos.mp4 
> _Win 11 (without TPM 2.0)_ _nvidia geforce GTX 1070_ _amd ryzen 5 1500X quad-core_ _Fly Photos 2.3.2_

Can you describe steps to reproduce this bug in detail? If I am able to reproduce the issue at my side, I can fix it.

**Comment by SABERBIRD** [comment]
Unfortunately, I cannot provide exact steps to reproduce the bug. However, I have the impression that the bug might be related to the cache.

A few times, the bug did NOT appear when opening the app, and once it did NOT show up on the first launch of the app after turning on the PC.

However, whenever I spent a lot of time scrolling through photos, the bug was always present.

**Comment by ccrowles** [comment]
> Unfortunately, I cannot provide exact steps to reproduce the bug. However, I have the impression that the bug might be related to the cache.
> 
> A few times, the bug did NOT appear when opening the app, and once it did NOT show up on the first launch of the app after turning on the PC.
> 
> However, whenever I spent a lot of time scrolling through photos, the bug was always present.

Can confirm that this is still the case. Sometimes the bug won't appear, but if you scroll images for a bit it does eventually happen.

**Comment by riyasy** [comment]
Sorry to hear this still persists. I have tried different OSs and systems to reproduce this bug and yet to see it. I shall review the code once again.

**Comment by riyasy** [comment]
Can you check with this version?
Uninstall previous version and install this version.

https://drive.google.com/file/d/1Odl-VKCshh9hq85J0shmIxPmedG_wxI_/view?usp=sharing

**Comment by ccrowles** [comment]
> Can you check with this version? Uninstall previous version and install this version.
> 
> https://drive.google.com/file/d/1Odl-VKCshh9hq85J0shmIxPmedG_wxI_/view?usp=sharing

It seems to work properly! I have tried scrolling through numerous transparent images and none of them seems to overlap so far, thank you!

**Comment by riyasy** [comment]
Fixed in 2.5.2
